src/Chip/ChipIcon.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { KeyboardEventHandler, MouseEventHandler } from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../Icon';
+// @ts-ignore
+import IconButton from '../IconButton';
+// @ts-ignore
+import { STYLE_VARIANTS } from './constants';
+
+export interface ChipIconProps {
+  className: string,
+  src: React.ReactElement | Function,
+  onClick?: KeyboardEventHandler & MouseEventHandler,
+  alt?: string,
+  variant: string,
+}
+
+function ChipIcon({
+  className, src, onClick, alt, variant,
+}: ChipIconProps) {
+  if (onClick) {
+    return (
+      <IconButton
+        className={className}
+        src={src}
+        onClick={onClick}
+        onKeyPress={onClick}
+        iconAs={Icon}
+        alt={alt}
+        invertColors={variant === STYLE_VARIANTS[1]}
+      />
+    );
+  }
+
+  return <Icon src={src} className={className} />;
```

Comment:
[curious] I'm wondering if we should be using the `size="sm"` attribute for the `Chip` icons so it more closely matches the small font size used inside the `Chip`? 

<img width="84" alt="image" src="https://github.com/openedx/paragon/assets/2828721/04948503-22ee-4a92-85aa-3411f1644827">